app/src/main/java/com/example/geoquiz/QuizViewModel.kt

```diff
@@ -0,0 +1,58 @@
+package com.example.geoquiz
+
+import androidx.lifecycle.ViewModel
+
+class QuizViewModel : ViewModel() {
+
+    var currentIndex: Int = 0
+    var isCheater = false
+    lateinit var correctAnswers: ArrayList<Int>
+    var messageResId: Int = 0
+
+    var questionBank = listOf<Question>(
+        Question(R.string.question_australia, true),
+        Question(R.string.question_oceans, true),
+        Question(R.string.question_mideast, false),
+        Question(R.string.question_africa, false),
+
+    )
+
+    var questionWasAnswered: BooleanArray = BooleanArray(questionBank.size)
+
+    val currentQuestionAnswer: Boolean
+        get() = questionBank[currentIndex].answerTrue
+
+    val currentQuestionText: Int
+        get() = questionBank[currentIndex].textResId
+
+    fun moveToNext() {
+        currentIndex = (currentIndex + 1) % questionBank.size
+    }
+
+    fun moveToPrev() {
+        currentIndex = (currentIndex - 1) % questionBank.size
```

Comment:
Is it necessary to have the % questionBank.size when you go down? Can you get above the questions bank size?